### PR TITLE
azure_rm_aks: cluster client & models API version

### LINKED ISSUE #489: [Feature Idea] Upgrade Managed Cluster Client Api Version To 2020-04-01
##### SUMMARY

Upgrade Managed Cluster Client Api Version To 2020-04-01 From The Current 2019-04-01

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

Components related to azure_rm_aks (Including info, upgrade_versions, etc.)

##### ADDITIONAL INFORMATION

This would allow for newer AKS features to be supported by this Galaxy Module

https://github.com/ansible-collections/azure/pull/336 currently relies on this API version to be upgraded

### PR DESCRIPTION
**Pull request by MarcGi**
##### SUMMARY

Fixes #489 
- Updated the AKS managed cluster client & models API version from 2019-04-01 to 2020-04-01
- Made required changes to properties based on API version.
- Updated the relevant documentation
- Added changeable properties to the update agent pool method that were previously missing
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- azure_rm_aks
- azure_rm_aks_info
- azure_rm_aksupgrade_info
##### ADDITIONAL INFORMATION

**Breaking Changes:**

  **azure_rm_aks**:
  - An agent_pool_profile now requires the 'mode' property
  
  **azure_rm_aksupgrade_info**:
  - upgrades.azure_aks_upgrades.agent_pool_profiles[n].kubernetes_version no longer returns an available upgrade version if the control plane needs updated first

Command Example:
```paste below
 - name: Create an AKS instance
   azure_rm_aks:
     name: "aks{{ rpfx }}"
     resource_group: "{{ resource_group }}"
     location: eastus
     dns_prefix: "aks{{ rpfx }}"
     kubernetes_version: "{{ versions.azure_aks_versions[0] }}"
     service_principal:
         client_id: "{{ azure_client_id }}"
         client_secret: "{{ azure_secret }}"
     linux_profile:
         admin_username: azureuser
         ssh_key: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDSPmiqkvDH1/+MDAVDZT8381aYqp73Odz8cnD5hegNhqtXajqtiH0umVg7HybX3wt1HjcrwKJovZURcIbbcDvzdH2bnYbF93T4OLXA0bIfuIp6M86x1iutFtXdpN3TTicINrmSXEE2Ydm51iMu77B08ZERjVaToya2F7vC+egfoPvibf7OLxE336a5tPCywavvNihQjL8sjgpDT5AAScjb3YqK/6VLeQ18Ggt8/ufINsYkb+9/Ji/3OcGFeflnDXq80vPUyF3u4iIylob6RSZenC38cXmQB05tRNxS1B6BXCjMRdy0v4pa7oKM2GA4ADKpNrr0RI9ed+peRFwmsclH test@ansible
     agent_pool_profiles:
         - name: default
           count: 1
           vm_size: Standard_B2s
           type: VirtualMachineScaleSets
           mode: System
           max_pods: 42
           availability_zones:
             - 1
             - 2
     node_resource_group: "node{{ noderpfx }}"
     enable_rbac: yes
     network_profile:
         load_balancer_sku: standard
   register: output
```

### CODE REVIEW

**Comment by MarcGi** [comment]
Hi @Fred-sun, Could you please review this when you have the time? Thank you very much!

**Comment by Fred-sun** [comment]
> Hi @Fred-sun, Could you please review this when you have the time? Thank you very much!


@MarcGi Thanks for your contribution! I'm working in!

File: plugins/modules/azure_rm_aks.py [hunk 1]
```diff
@@ -267,9 +276,19 @@
...
             count: 1
-            vm_size: Standard_DS1_v2
+            vm_size: Standard_B2s
             enable_auto_scaling: True
             type: VirtualMachineScaleSets
+            mode: System
+            max_count: 3
+            min_count: 1
+            enable_rbac: yes
+         - name: user
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: Examples is not valid YAML.
```suggestion
          - name: user
```

Comment (MarcGi) [reply plugins/modules/azure_rm_aks.py]: I have committed this change, thanks!

File: plugins/modules/azure_rm_aks.py [hunk 2]
```diff
@@ -99,6 +99,15 @@
                     - 'VirtualMachineScaleSets'
                     - 'AvailabilitySet'
                 type: str
+            mode:
+                description:
+                    - AgentPoolMode represents mode of an agent pool.
+                    - Possible values include C(System) and C(User).
+                    - System AgentPoolMode requires a minimum VM SKU of at least 2 vCPUs and 4GB memory
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]:
```suggestion
                    - System AgentPoolMode requires a minimum VM SKU of at least 2 vCPUs and 4GB memory.
```

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**